tokio-quiche/src/quic/io/worker.rs

```diff
@@ -529,6 +637,69 @@ where
         }
     }
 
+    fn sync_connection_ids(
+        &mut self, qconn: &mut QuicheConnection,
+    ) -> QuicResult<()> {
+        let mut current_cids = HashSet::new();
+
+        // Process all current source IDs
+        for cid in qconn.source_ids() {
+            let owned_cid = cid.clone().into_owned();
+            current_cids.insert(owned_cid.clone());
+
+            // Only send MapCid if this is a new CID we haven't mapped before
+            if !self.known_cids.contains(&owned_cid) {
+                log::trace!(
+                    "Mapping new SCID {:?} to connection ID {}",
```

Comment:
Nit: this is the worker id, not the connection ID. The trace ID of the connection can be obtained using `qconn.trace_id()`. Also, it may make sense to prefix all Worker-related messages with the worker ID for debugging processing purposes.
```suggestion
                    "Mapping new SCID {:?} to worker ID {}",
```